tests/test_worker.py

```diff
@@ -12,27 +12,112 @@
+    def test_lifecycle_methods_shell_out_to_execute(
+            self, execute_mock, unused_configure_mock):
         presto_worker = Worker()
+
         presto_worker.install(self.mock_env)
+        assert execute_mock.called
+        execute_mock.reset_mock()
+
+        presto_worker.stop(self.mock_env)
+        assert execute_mock.called
+        execute_mock.reset_mock()
+
+        presto_worker.start(self.mock_env)
+        assert execute_mock.called
+        execute_mock.reset_mock()
+
+        presto_worker.status(self.mock_env)
+        assert execute_mock.called
+
+    @patch('package.scripts.presto_worker.Worker.configure')
+    @patch('package.scripts.presto_worker.Execute')
+    def test_install_start_configure_presto(
+            self, unused_execute_mock, configure_mock):
+        presto_worker = Worker()
+
+        presto_worker.install(self.mock_env)
+        assert configure_mock.called
+        configure_mock.reset_mock()
+
+        presto_worker.start(self.mock_env)
+        assert configure_mock.called
+
+    @patch('package.scripts.presto_worker.create_tpch_connector')
+    def test_configure_adds_tpch_connector(self, create_tpch_connector_mock):
+         presto_worker = Worker()
+
+         with patch('__builtin__.open'):
+            presto_worker.configure(self.mock_env)
+
+         assert create_tpch_connector_mock.called
+
+    @patch('package.scripts.presto_worker.create_tpch_connector')
+    @patch('package.scripts.params.config_properties', new=dummy_config_properties)
+    def test_configure_ignore_pseudo_distribute_enabled_property(self, create_tpch_connector_mock):
+        config = []
+        open_mock = mock_file_descriptor_write_method(config)
+        presto_worker = Worker()
+
+        with patch('__builtin__.open', open_mock):
+            presto_worker.configure(self.mock_env)
+
+        assert 'pseudo.distributed.enabled=true\n' not in config
+
+    @patch('package.scripts.presto_worker.create_tpch_connector')
+    @patch('package.scripts.params.config_properties', new=dummy_config_properties)
+    def test_configure_ignore_empty_queue_config_file(self, create_tpch_connector_mock):
+        config = []
+        open_mock = mock_file_descriptor_write_method(config)
+        presto_worker = Worker()
+
+        with patch('__builtin__.open', open_mock):
+            presto_worker.configure(self.mock_env)
+
+        for item in config:
+            assert not item.startswith('query.queue-config-file')
+
+    @patch('package.scripts.presto_worker.create_tpch_connector')
+    @patch('package.scripts.params.config_properties', new=dummy_config_properties)
+    def test_config_properties_coordinator_always_false(self, create_tpch_connector_mock):
+        config = []
+        open_mock = mock_file_descriptor_write_method(config)
+        presto_worker = Worker()
+
+        with patch('__builtin__.open', open_mock):
+            presto_worker.configure(self.mock_env)
```

Comment:
above 5 lines of code is repeated in several tests, maybe extract this as method?